x-pack/plugins/task_manager/server/task_running/task_runner.ts

```diff
@@ -261,7 +269,17 @@ export class TaskManagerRunner implements TaskRunner {
 
     try {
       this.task = this.definition.createTaskRunner(modifiedContext);
-      const result = await withSpan({ name: 'run', type: 'task manager' }, () => this.task!.run());
+      const ctx = {
+        type: 'task manager',
+        name: `run ${this.instance.task.taskType}`,
+        id: this.instance.task.id,
+        description: 'run a task',
+        // is there a way to create a URL to navigate to a task
+        // url: '',
+      };
+      const result = await this.executionContext.withContext(ctx, () =>
+        withSpan({ name: 'run', type: 'task manager' }, () => this.task!.run())
```

Comment:
The concept is quite similar to apm agent instrumentation. Maybe in the future, we will even use `apm-nodejs-agent` under the hood to attach the context data to the current transaction.

Reply:
Ya, that would actually be pretty nice.